snapcraft/application.py

```diff
@@ -22,21 +22,70 @@
 import pathlib
 import signal
 import sys
-from typing import Any
+from typing import Any, List, cast
 
 import craft_application.commands as craft_app_commands
+import craft_application.models
 import craft_cli
 from craft_application import Application, AppMetadata, util
+from craft_application.models import BuildInfo
 from craft_cli import emit
+from craft_providers import bases
 from overrides import override
 
 from snapcraft import cli, errors, models, services
 from snapcraft.commands import unimplemented
+from snapcraft.models import Architecture
+from snapcraft.providers import SNAPCRAFT_BASE_TO_PROVIDER_BASE
+from snapcraft.utils import get_effective_base, get_host_architecture
+
+
+class SnapcraftBuildPlanner(craft_application.models.BuildPlanner):
+    """A project model that creates build plans."""
+
+    def get_build_plan(self) -> List[BuildInfo]:
```

Comment:
Should `snapcraft.models.project.Project.get_build_plan()` be removed as part of this PR?  Or is this an "add-before-remove" strategy?

Reply:
Removing it caused some breakages - I think we need some upstream changes to fix that.

Reply:
please create a task for this

Reply:
https://github.com/canonical/craft-application/issues/208